pkg/grpc/api.go

```diff
@@ -787,6 +795,119 @@ func (a *api) getStateStore(name string) (state.Store, error) {
 	return a.stateStores[name], nil
 }
 
+func (a *api) getComponent(componentKind string, componentName string) (component interface{}) {
+	switch componentKind {
+	case "state":
+		if statestore := a.stateStores[componentName]; statestore != nil {
+			component = statestore
+		}
+	case "pubsub":
+		if pubsub := a.pubsubAdapter.GetPubSub(componentName); pubsub != nil {
+			component = pubsub
+		}
+	case "secretstores":
+		if secretstore := a.secretStores[componentName]; secretstore != nil {
+			component = secretstore
+		}
+	case "bindings":
+		if inputbinding := a.inputBindings[componentName]; inputbinding != nil {
+			component = inputbinding
+		} else if outputbinding := a.outputBindings[componentName]; outputbinding != nil {
+			component = outputbinding
+		}
+	}
+	return
+}
+
+// GetComponentHealthAlpha1 returns the health of components.
+// If a componentName is specified, it returns the health of that component.
+// If no componentName is specified, it returns the health of all components.
+func (a *api) GetComponentHealthAlpha1(ctx context.Context, in *runtimev1pb.ComponentHealthRequest) (*runtimev1pb.ComponentHealthResponse, error) {
+	componentName := in.ComponentName
+	components := a.getComponentsFn()
+	if componentName == nil || *componentName == "" {
+		return a.getAllComponentsHealth(ctx)
+	}
+	for _, comp := range components {
+		if *componentName == comp.Name {
+			status, _, _, err := a.checkHealthUtil(strings.Split(comp.Spec.Type, ".")[0], comp.Name)
```

Comment:
I tried using SplitN but in the last parameter, it requires a parameter telling output slice capacity. Now, this may vary. As of now, I am sticking with Split only. It seems straight-forward.

Reply:
No, it just limits the splitting. In your case, you'd use `strings.SplitN(comp.Spec.Type, ".", 2)[0]` since you're only interested in the first part.

Reply:
Agreed, this is better - I tried with something like pubsub.kafka and bindings.azure.eventgrid - these work correct - changed.